### PR TITLE
Featured communities component

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Featured communities component displays the 7 Digital.gov Communities of Practice with an emphasized community card in the first position.

<details><summary>Default (Home Page)</summary>
Screen Shot 2023-12-05 at 2 56 32 PM
</details>

<details><summary>Has-Image (Communities Page)</summary>
Screen Shot 2023-12-05 at 2 56 43 PM
</details>

### Preview
- [Home Page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-featured-component/)
- [Communities Page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-featured-component/communities)

### Solution

1. `layout` setting set styles depending on page
	- `has-image` variant will display brand logo image and emphasize first card button style with brand color
3. Lookup `communities/_index.md` to set the header `title` and `deck` dynamically.
4. Shortened the copy for each community `summary` section for smaller card height sizes. See changes below
5. Set `weight: 2` for `web-managers-forum` to position as first card.
6. Removed unnecessary grid classes on `communities/list.html`
7. Moved usa-collection for "Other Communities" to it's own partial

**Community Summary Changes**

| community                      | original                                                                                                                                                             | shortened                                                                                   |
| ------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
| communicators                  | Brings together government communicators to improve communication and make government more effective.                                                                | Improve written and verbal communication to make government more effective.                 |
| multilingual                   | Brings together government multilingual content managers to expand and improve digital content in languages other than English.                                      | Expand and improve digital content in languages other than English.                         |
| plain-language                 | Brings together government plain language practitioners to promote use of plain language, saving agencies time and money and providing better service to the public. | Promote the use of plain language to provide better service to the public.                  |
| social-media                   | Brings together government social media communicators to improve social media outreach efforts.                                                                      | Improve social media outreach and performance.                                              |
| user-experience                | Brings together government user experience practitioners to create better user-centered products for the public.                                                     | Create better user-centered products for the public.                                        |
| web-analytics-and-optimization | Brings together government analytics practitioners to make better decisions using web analytics and other optimization strategies.                                   | Make better decisions using web analytics and other optimization strategies.                |
| web-content-managers           | Brings together government web content managers to create a better online experience for the public.                                                                 | Create a better online experience for the public alongside government web content managers. | 

### How To Test

1. Visit [homepage](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-featured-component/) and [communities](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-featured-component/communities) landing page
2. Card styles are different for each page

### What To Test
- [ ] first card is displayed with width of 2 cards
- [ ] default component (home page) displays centered text, button background color and no image.
- [ ] has-media variant (communities list page) displays left justified text, image and button outline styles.

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

File: themes/digital.gov/src/scss/new/home/_communities.scss [hunk 1]
```diff
@@ -43,3 +44,74 @@ $communities-of-practice: (
...
+        }
+      }
+
+      .usa-button--outline {
+        box-shadow: inset 0 0 0 2px map.get($communities-of-practice, "WCM");
+        color: map.get($communities-of-practice, "WCM");
+        &:hover {
+          color: #fff;
+        }
+      }
```
Comment (nick-mon1) [inline themes/digital.gov/src/scss/new/home/_communities.scss:82]: Styles are still coupled to `WCM` brand colors and will need to be updated when changing the weight for a featured community card.

File: themes/digital.gov/layouts/communities/list.html [hunk 2]
```diff
@@ -1,123 +1,27 @@
...
-                </h2>
-                <p class="usa-collection__description">
-                  {{- $e.Params.summary | markdownify -}}
-                </p>
-              </div>
-            </li>
-          {{ end }}
-        </ul>
-      </div>
+      {{ partial "core/collection/collection-grid-list.html" (dict "list_data" $other_communities) }}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/communities/list.html:24]: Moved this collection related markup to it's own partial in the collection partials folder.

File: content/communities/communicators.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 slug: communicators
 date: 2021-02-24 18:49:00 -0500
 title: Communicators
-summary: "Brings together government communicators to improve communication and make government more effective."
+summary: "Improve communication and make government more effective."
```
Comment (bonnieAcameron) [inline content/communities/communicators.md:5]:
```suggestion
summary: "Improve communication to make government more effective."
```

File: content/communities/plain-language-community-of-practice.md [hunk 4]
```diff
@@ -5,7 +5,7 @@
 slug: plain-language
 date: 2016-02-03 10:58:53 -0400
 title: "Plain Language"
-summary: "Brings together government plain language practitioners to promote use of plain language, saving agencies time and money and providing better service to the public."
+summary: "Promote use of plain language, providing better service to the public."
```
Comment (bonnieAcameron) [inline content/communities/plain-language-community-of-practice.md:8]:
```suggestion
summary: "Promote the use of plain language to provide better service to the public."
```

File: content/communities/social-media.md [hunk 5]
```diff
@@ -5,7 +5,7 @@
 slug: social-media
 date: 2019-12-12 19:00:00 -0500
 title: "Social Media"
-summary: "Brings together government social media communicators to improve social media outreach efforts."
+summary: "Improve social media outreach efforts."
```
Comment (bonnieAcameron) [inline content/communities/social-media.md:8]:
```suggestion
summary: "Improve social media outreach and performance."
```

File: content/communities/web-managers-forum.md
Comment (bonnieAcameron) [inline content/communities/web-managers-forum.md:1]: Suggestion to be more parallel with your other community descriptions: 

Line 8: 
`summary: "Create a better online experience for the public alongside government web content managers."
`

**Review by bonnieAcameron — CHANGES_REQUESTED** [review]
Wording suggestions

File: content/communities/communicators.md (hunk 3 above)
Comment (ToniBonittoGSA) [reply content/communities/communicators.md:5]: @nick-mon1 pls make that: 
"Improve written and verbal communication to make ..."

File: content/communities/plain-language-community-of-practice.md (hunk 4 above)
Comment (ToniBonittoGSA) [reply content/communities/plain-language-community-of-practice.md:8]: Agreed

File: content/communities/social-media.md (hunk 5 above)
Comment (ToniBonittoGSA) [reply content/communities/social-media.md:8]: Agreed

**Comment by ToniBonittoGSA** [comment]
@nick-mon1 Looks great! Can you make the rest of the button hovers appear like the Web Managers one? The color matches the swatch at the top

**Review by mejiaj — CHANGES_REQUESTED** [review]
Looks really good so far. Pushed some updates to simplify markup, remove unused styles, consistency in styles. 

Only one outstanding comment on logic.

**Review by bonnieAcameron — APPROVED** [review]

**Review by clmedders — APPROVED** [review]

File: themes/digital.gov/layouts/partials/core/featured-communities.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+{{ $context := .context }}
+{{ $layout := .layout }}
+
+{{ $page_header := $context.Site.GetPage "/communities" }}
+{{ $communities := (where $context.Site.Pages "Section" "communities").ByTitle }}
+{{ $communities = (where $communities ".Params.weight" "ge" 1) }}
+{{ $dg_communities := (where $communities ".Params.dg_highlight" "eq" true) }}
+{{/* featured community is listed first, followed by the other dg 7 listed alphabetically by title */}}
+{{ $featured_community := (where $communities ".Params.weight" "eq" 2) }}
+{{ $dg_communities = union $featured_community $dg_communities }}
```
[the remark this answers is not in the document]
Comment (mejiaj) [reply themes/digital.gov/layouts/partials/core/featured-communities.html:18]: Just checking in, was this comment addressed/resolved?

**Review by mejiaj — CHANGES_REQUESTED** [review]
Can we resolve / address comments?

If they're not major and within the scope of this PR we can also just say that and resolve the thread.

File: themes/digital.gov/layouts/communities/list.html [hunk 7]
```diff
@@ -1,123 +1,23 @@
...
-          </header>
-        </div>
+  {{/* Gets all community pages and filter out those that are not the featured dg top 7 */}}
+  {{ $communities := (where (where .Site.Pages "Section" "communities").ByTitle ".Params.weight" "ge" 1) }}
+  {{ $other_communities := (where $communities ".Params.dg_highlight" "!=" true) }}
+
+
+  <main class="grid-container-desktop" id="main-content">
+    {{/* Digital.gov Communities of Practice */}}
+    {{ partial "core/featured-communities.html" (dict "context" . "layout" "has-image") }}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/communities/list.html:9]: For displaying images, left justified text and outline button styles, use the `has-image` variant.

```
{{ partial "core/featured-communities.html" (dict "context" . "layout" "has-image") }}
``` 

Default component does not need this setting to display no image, centered text and default button styles.

File: themes/digital.gov/layouts/partials/core/featured-communities.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/featured-communities.html:18]: Hey @afeijoo or @ToniBonittoGSA,

Just wanted to confirm we want `Web Managers Forum` first for the community section?
If not, we can just go with the simpler option.

File: themes/digital.gov/layouts/partials/core/usa-card.html [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+{{- $link_title := default "" .card_data.link_title -}}
+{{- $image := default "" .card_data.image -}}
+{{- $alt := default "" .card_data.alt -}}
+{{- $path := default "" .card_data.path -}}
+{{- $layout := default "" .card_data.layout -}}
+{{- $has_image := false -}}
+
+{{ if eq $layout "has-image" }}
+  {{ $has_image = true }}
+{{ end }}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/partials/core/usa-card.html:26]: Updated logic to set default `{{- $has_image := false -}}`.

File: themes/digital.gov/src/scss/new/home/_communities.scss (hunk 1 above)
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/home/_communities.scss]: Just the first card in the list.

File: themes/digital.gov/layouts/partials/core/featured-communities.html [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  - "has-media" variant displays an image, full background colors on buttons, and left justified copy
+*/}}
+
+{{ $context := .context }}
+{{ $layout := .layout }}
+
+{{ $page_header := $context.Site.GetPage "/communities" }}
+{{ $communities := (where $context.Site.Pages "Section" "communities").ByTitle }}
+{{ $communities = (where $communities ".Params.weight" "ge" 1) }}
+{{ $dg_communities := (where $communities ".Params.dg_highlight" "eq" true) }}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/partials/core/featured-communities.html:15]: Simplified logic as per James's request. Displays alphabetically by default.

**Review by mejiaj — DISMISSED** [review]
Looks good. 

Just noting that the Web Managers community is no longer highlighted. If necessary, we can do a quick follow-up as to not hold up this work.

**Screenshot**
image

**Review by clmedders — APPROVED** [review]

**Pull request merged**